doc/styleguide.md

```diff
@@ -0,0 +1,248 @@
+```
+
+For API references, there are exceptions to this rule.
+
+## Markdown rules
+
+This repository uses the [`remark-preset-lint-node`][remark-preset-lint-node]
+package to enforce consistent Markdown styling. For the exact rules, see
+[nodejs/remark-preset-lint-node][nodejs/remark-preset-lint-node]. Additionally,
+we're beginning to use [docs-parser][docs-parser] to parse the Markdown
+into an API.
+
+There are a few style guidelines that aren't covered by the linter rules:
+
+* Use `sh` instead of `cmd` in code blocks (due to the syntax highlighter).
+* Keep line lengths between 80 and 100 characters if possible for readability
+  purposes.
+* No nesting lists more than 2 levels (due to the markdown renderer).
+* All `js` and `javascript` code blocks are linted with
+  [standard-markdown](https://www.npmjs.com/package/standard-markdown).
+* For unordered lists, use asterisks instead of dashes.
+
+## Picking words
+
+* Use "will" over "would" when describing outcomes.
+
+## API references
+
+The following rules only apply to the documentation of APIs.
+
+### Title and description
+
+Each module's API doc must use the actual object name returned by requiring it
+as its title (such as `path`, `fs`, and `querystring`).
+
+Directly under the page title, add a one-line description of the module
+as a markdown quote (beginning with `>`).
+
+Using the `session` module as an example:
+
+```markdown
+# querystring
+
+> Utilities for parsing and formatting URL query strings.
+```
+
+### Module methods and events
+
+For modules that are not classes, their methods and events must be listed under
+the `## Methods` and `## Events` chapters.
+
+Using `fs` as an example:
+
+```markdown
+# fs
+
+## Events
+
+### Event: 'close'
+
+## Methods
+
+### `fs.access(path[, mode], callback)`
+```
+
+### Classes
+
+* API classes or classes that are part of modules must be listed under a
+  `## Class: TheClassName` chapter.
+* One page can have multiple classes.
+* Constructors must be listed with `###`-level headings.
+* [Static Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/static)
+  must be listed under a `### Static Methods` chapter.
+* [Instance Methods](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Prototype_methods)
+  must be listed under an `### Instance Methods` chapter.
+* All methods that have a return value must start their description with
+  ``Returns `[TYPE]` - (Return description)``
+  * If the method returns an `Object`, its structure can be specified
+    using a colon followed by a newline then an unordered list of properties in
+    the same style as function parameters.
+* Instance Events must be listed under an `### Instance Events` chapter.
+* Instance Properties must be listed under an `### Instance Properties` chapter.
+  * Instance Properties must start with `A [Property Type] ...`
+
+Using the `v8` classes as an example of some of the outlined structure:
+
+```markdown
+# v8
```

Comment:
I wonder if the H1s here are supposed to be in title case &mdash; if so, V8 is a proper noun and would need to be capitalized.

Reply:
> I wonder if the H1s here are supposed to be in title case — if so, V8 is a proper noun and would need to be capitalized.

This seems to refer to the `v8` module in Node.js and not the V8 JavaScript engine, so it should not be capitalized but it probably should be in backticks to indicate that it is code text and not generic prose. (I'm OK if the resultant HTML doesn't render it differently due to the backticks. It should still be there in markdown for clarity around issues like this one.)

```suggestion
# `v8`
```